lib/buffer.js

```diff
@@ -272,19 +272,66 @@ Buffer.concat = function(list, length) {
 };
 
 
+function base64ByteLength(str) {
+  var len = str.length;
+  var bytes = len;
+
+  // Handle padding
+  if (str[len - 1] === '=')
+    bytes--;
+  if (len > 2 && str[len - 2] === '=')
+    bytes--;
+
+  // Base64 ratio: 3/4
+  bytes = (bytes / 4) * 3;
```

Comment:
Maybe `bytes = (bytes >>> 2) * 3`?  Or if you're feeling fancy:

```javascript
bytes >>>= 2;
bytes += bytes + bytes;
```

Although it wouldn't surprise me if V8 already does the multiplication optimization.

(And besides, my Core i7 does an imul in 5 cycles so it's not as if it's tremendously expensive.)